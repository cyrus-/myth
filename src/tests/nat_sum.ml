let s = Nat_decls.s ^ "

let nat_add : nat -> nat -> nat |>
  { 0 => ( 0 => 0
         | 1 => 1
         | 2 => 2 )
  | 1 => ( 0 => 1
         | 1 => 2
         | 2 => 3 )
  | 2 => ( 0 => 2
         | 1 => 3
         | 2 => 4 )
  } = ?"
