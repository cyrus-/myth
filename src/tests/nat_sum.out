(* Generated Thu Nov 13 00:35:52 2014 *)
let nat_add : nat -> nat -> nat =
  let rec f1 (n1:nat) : nat -> nat =
    fun (n2:nat) -> match n1 with
                      | O -> n2
                      | S (n3) -> S (f1 n3 n2)
  in
    f1
;;
